stdlib/statistics.pyi

```diff
@@ -32,8 +32,8 @@ if sys.version_info >= (3, 13):
     __all__ += ["kde", "kde_random"]
 
 # Most functions in this module accept homogeneous collections of one of these types
-_Number: TypeAlias = float | Decimal | Fraction
-_NumberT = TypeVar("_NumberT", float, Decimal, Fraction)
+_Number: TypeAlias = Real | Decimal
```

Comment:
I'm afraid type checkers don't understand float to be a subtype of numbers.Real

Reply:
I'm a bit scared now — I know pyre does, and ISTR mypy did as well?

Is there an easy-to-use playground to run the same typing information across all the supported checkers?

Reply:
Mypy has never properly supported the ABCs in the numbers module, and I don't believe pyright has either. There's no cross-typechecker playground, but MyPy's playground is at https://mypy-play.net/?mypy=latest&python=3.12 and pyright's is at https://pyright-play.net/

Reply:
You can see evidence of this in the new `openlibrary` hit reported by mypy_primer

Reply:
Okay this is worse than I expected :(

Not only mypy doesn't support it (I *might* be confusing it with pytype, but it's been years, I'm lost), but neither of them seem to even think that a `* 2` maintains the type.

Do you have a suggestion on how to handle this? Just add `Real` without removing `float` ? Replace it with `SupportsFloat` even if it's not quite the right type?

Reply:
The numbers ABCs are sort-of fundamentally broken when it comes to static typing, I'm afraid. It's impossible to add proper annotations to many of the methods for those ABCs in numbers.pyi here in typeshed, and so as a result we deliberately don't have the builtin numeric classes inherit from them in builtins.pyi; this is why type checkers do not understand them.

I'll take a look in a bit and see if I can think of what to do here

Reply:
Can I ask what the problem you're trying to solve is here? I feel like our current versions of `_Number` and `_NumberT` here are going to work well for the vast majority of uses of the `statistics` functions. Is there a specific false positive mypy emitted on some code that you're trying to fix?